{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sys\n",
    "import boto3\n",
    "import os\n",
    "import json\n",
    "import mysql.connector\n",
    "\n",
    "with open('config.json') as f:\n",
    "    creds = json.load(f)\n",
    "\n",
    "\n",
    "\n",
    "ENDPOINT=creds['endpoint']\n",
    "PORT=creds['port']\n",
    "USR=creds['user']\n",
    "REGION=creds['region']\n",
    "DBNAME=creds['dbname']\n",
    "os.environ['LIBMYSQL_ENABLE_CLEARTEXT_PLUGIN'] = '1'\n",
    "\n",
    "#gets the credentials from .aws/credentials\n",
    "session = boto3.Session(profile_name='default')\n",
    "client = session.client('rds')\n",
    "\n",
    "token = client.generate_db_auth_token(DBHostname=ENDPOINT, Port=PORT, DBUsername=USR, Region=REGION)                    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'discord-bot-vault.ca24phzvaaar.us-east-2.rds.amazonaws.com:3306/?Action=connect&DBUser=admin&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAJINX7JG5T7JQWHWQ%2F20211130%2Fus-east-2%2Frds-db%2Faws4_request&X-Amz-Date=20211130T043932Z&X-Amz-Expires=900&X-Amz-SignedHeaders=host&X-Amz-Signature=44fa66d7f02f87abd2ea54f7297d8bc14747e5efb93f6a75c64a9688568bc6cd'"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "token"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Database connection failed due to 2005 (HY000): Unknown MySQL server host 'discord-bot-vault.ca24phzvaaar.us-east-2.rds.amazonaws.com' (0)\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    conn =  mysql.connector.connect(host=ENDPOINT, user=USR, passwd=token, port=PORT, database=DBNAME)\n",
    "    cur = conn.cursor()\n",
    "    cur.execute(\"\"\"SELECT now()\"\"\")\n",
    "    query_results = cur.fetchall()\n",
    "    print(query_results)\n",
    "except Exception as e:\n",
    "    print(\"Database connection failed due to {}\".format(e)) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "rds = boto3.client('rds')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'Address': 'aws-fin-db.ca24phzvaaar.us-east-2.rds.amazonaws.com', 'Port': 3306, 'HostedZoneId': 'Z2XHWR1WZ565X2'}\n",
      "admin\n",
      "{'Address': 'discord-bot-vault.ca24phzvaaar.us-east-2.rds.amazonaws.com', 'Port': 3306, 'HostedZoneId': 'Z2XHWR1WZ565X2'}\n",
      "admin\n"
     ]
    }
   ],
   "source": [
    "try:\n",
    "    dbs = client.describe_db_instances()\n",
    "    for db in dbs['DBInstances']:\n",
    "        print(db['Endpoint'])\n",
    "        print(db['MasterUsername'])\n",
    "except Exception as e:\n",
    "    raise(e)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
